File: java-samples/Advanced/duediligence-cordapp/contracts/src/test/java/net/corda/samples/duediligence/contracts/StateTests.java
package net.corda.samples.duediligence.contracts;

import net.corda.core.identity.Party;
import net.corda.samples.duediligence.states.CorporateRecordsAuditRequest;
import net.corda.testing.node.MockServices;
import org.junit.Test;

public class StateTests {
    private final MockServices ledgerServices = new MockServices();

    @Test
    public void hasFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        CorporateRecordsAuditRequest.class.getDeclaredField("applicant");
        // Is the message field of the correct type?
        assert(CorporateRecordsAuditRequest.class.getDeclaredField("applicant").getType().equals(Party.class));
    }
}

File: kotlin-samples/Advanced/duediligence-cordapp/contracts/src/test/kotlin/net/corda/samples/duediligence/contracts/StateTests.kt
package net.corda.samples.duediligence.contracts

import net.corda.core.identity.Party
import net.corda.samples.duediligence.states.CorporateRecordsAuditRequest
import org.junit.Test
import kotlin.test.assertEquals

class StateTests {
    @Test
    fun hasFieldOfCorrectType() {
        // Does the amount field exist?
        CorporateRecordsAuditRequest::class.java.getDeclaredField("applicant")
        // Is the amount field of the correct type?
        assertEquals(CorporateRecordsAuditRequest::class.java.getDeclaredField("applicant").type, Party::class.java)
    }
}